applications/OptimizationApplication/python_scripts/output/container_variable_data_holder_output_decorator.py

```diff
@@ -0,0 +1,228 @@
+
+    def ResetData(self):
+        for data in self.data_list:
+            data["temp"].AssignDataToContainerVariable(data["variable"])
+            del data["temp"]
+
+    @staticmethod
+    def GetContainerType(container: ContainerVariableDataHolderUnion) -> str:
+        if isinstance(container, KratosOA.NodalContainerVariableDataHolderBase):
+            return "nodes"
+        elif isinstance(container, KratosOA.ConditionContainerVariableDataHolderBase):
+            return "condition"
+        elif isinstance(container, KratosOA.ElementContainerVariableDataHolderBase):
+            return "elements"
+        else:
+            raise RuntimeError(f"The container type cannot be determined. [ Container = {str(container)}]")
+
+class ResponseContainerVariableDataHolderInfo(ContainerVariableDataHolderInfo):
+    @classmethod
+    def GetDefaultParameters(cls) -> Kratos.Parameters:
+        return Kratos.Parameters("""{
+            "type"           : "response_data",
+            "model_part_name": "PLEASE_PROVIDE_MODEL_PART_NAME",
+            "container_type" : "PLEASE_PROVIDE_CONTAINER_TYPE",
+            "response_name"  : "PLEASE_PROVIDE_RESPONSE_FUNCTION_NAME",
+            "data_list"      : [
+                {
+                    "data_name"     : "PLEASE_PROVIDE_DATA_NAME",
+                    "variable_name" : "PLEASE_PROVIDE_VARIABLE_NAME"
+                }
+            ]
+        }""")
+
+    @classmethod
+    def GetPattern(cls) -> str:
+        return "problem_data/response_data/<model_part_name>/<response_name>/sensitivities"
+
+class ControlContainerVariableDataHolderInfo(ContainerVariableDataHolderInfo):
+    @classmethod
+    def GetDefaultParameters(cls) -> Kratos.Parameters:
+        return Kratos.Parameters("""{
+            "type"           : "control_data",
+            "model_part_name": "PLEASE_PROVIDE_MODEL_PART_NAME",
+            "container_type" : "PLEASE_PROVIDE_CONTAINER_TYPE",
+            "control_name"   : "PLEASE_PROVIDE_CONTROL_NAME",
+            "data_list"      : [
+                {
+                    "data_name"     : "PLEASE_PROVIDE_DATA_NAME",
+                    "variable_name" : "PLEASE_PROVIDE_VARIABLE_NAME"
+                }
+            ]
+        }""")
+
+    @classmethod
+    def GetPattern(cls) -> str:
+        return "problem_data/control_data/<model_part_name>/<control_name>"
+
+class ContainerVariableDataHolderOutputDecorator(Kratos.OutputProcess):
+    def __init__(self, model: Kratos.Model, parameters: Kratos.Parameters, optmization_info: OptimizationInfo):
+        super().__init__()
+
+        default_parameters = Kratos.Parameters("""{
+            "module"                           : "KratosMultiphysics.OptimizationApplication.utilities",
+            "type"                             : "ContainerVariableDataHolderOutputDecorator",
+            "output_processes"                 : [],
+            "optimization_info_output_settings": []
+        }""")
```

Comment:
I think `Process` has `GetDefaultParameters` so I think it's better to override it rather than defining it in local scope.